consult.el

```diff
@@ -3724,35 +3730,40 @@ This command can act as a drop-in replacement for `repeat-complex-command'."
 
-If the minibuffer is active, returns the minibuffer history,
-otherwise the history corresponding to the mode is returned.
-There is a special case for `repeat-complex-command',
-for which the command history is used."
-  (cond
-   (history)
-   ;; If pressing "C-x M-:", i.e., `repeat-complex-command',
-   ;; we are instead querying the `command-history' and get a full s-expression.
-   ;; Alternatively you might want to use `consult-complex-command',
-   ;; which can also be bound to "C-x M-:"!
-   ((eq last-command 'repeat-complex-command)
-    (setq history (mapcar #'prin1-to-string command-history)))
-   ;; In the minibuffer we use the current minibuffer history,
-   ;; which can be configured by setting `minibuffer-history-variable'.
-   ((minibufferp)
-    (if (eq minibuffer-history-variable t)
-        (user-error "Minibuffer history is disabled for `%s'" this-command)
-      (setq history (mapcar #'consult--tofu-hide (symbol-value minibuffer-history-variable)))))
-   ;; Otherwise we use a mode-specific history, see `consult-mode-histories'.
-   (t (when-let (found
-                 (or (seq-find (lambda (ring)
-                                 (and (derived-mode-p (car ring))
-                                      (boundp (cdr ring))))
-                               consult-mode-histories)
-                     (user-error
-                      "No history configured for `%s', see `consult-mode-histories'"
-                      major-mode)))
-        (setq history (symbol-value (cdr found))))))
-  (consult--remove-dups (if (ring-p history) (ring-elements history) history)))
+Returns the cons (HISTORY . INDEX).  If the minibuffer is active,
+HISTORY is the minibuffer history, otherwise the history
+corresponding to the mode.  There is a special case for
+`repeat-complex-command', for which the command history is used.
+INDEX is the name of the index variable to update, if any."
+  (let (index)
+    (cond
+     (history)
+     ;; If pressing "C-x M-:", i.e., `repeat-complex-command',
+     ;; we are instead querying the `command-history' and get a full s-expression.
+     ;; Alternatively you might want to use `consult-complex-command',
+     ;; which can also be bound to "C-x M-:"!
+     ((eq last-command 'repeat-complex-command)
+      (setq history (mapcar #'prin1-to-string command-history)))
+     ;; In the minibuffer we use the current minibuffer history,
+     ;; which can be configured by setting `minibuffer-history-variable'.
+     ((minibufferp)
+      (if (eq minibuffer-history-variable t)
+	  (user-error "Minibuffer history is disabled for `%s'" this-command)
+	(setq history (mapcar #'consult--tofu-hide (symbol-value minibuffer-history-variable)))))
+     ;; Otherwise we use a mode-specific history, see `consult-mode-histories'.
+     (t (when-let (found
+		   (or (seq-find (lambda (ring)
+				   (and (derived-mode-p (car ring))
+					(boundp (let (c (cdr ring)) (if (consp c) (car c) c)))))
```

Comment:
I got some compilation warnings for this line. Do you use package-lint?

Reply:
Oops looks like I forgot to double wrap the let vars...  Do you run a linter on save?  Never set that up.

Reply:
I have the linter always on via flymake. In every Elisp buffer, also in my init.el. It is quite helpful.

Reply:
What I should probably set up also is some CI for github to check PRs.